src/NzbDrone.Core/Download/DownloadEventHub.cs

```diff
@@ -35,6 +36,14 @@ public void Handle(DownloadFailedEvent message)
             RemoveFromDownloadClient(trackedDownload);
         }
 
+        public void Handle(DownloadImportingEvent message)
+        {
+            if (message.TrackedDownload.DownloadItem.OutputPath.IsEmpty)
```

Comment:
I'm thinking that maybe using the boolean I mentioned earlier is still a good idea. Like `TorrentDetailsDeferred`.
Then you have a `downloadClient.PopulateTorrentDetails(trackedDownload.DownloadItem)` method doing the actual work.

`downloadClient.PopulateTorrentDetails(trackedDownload.DownloadItem)`

That way it's higher abstraction and in the future you can do more 'deferred' stuff. Similar to the `MarkAsImported` function which atm just sets the PostImportCategory.